dtfs-central-api/src/services/tfm/deal-cancellation.service.processPendingCancellation.test.ts

```diff
@@ -55,26 +67,33 @@ describe('DealCancellationService', () => {
   });
 
   describe('processPendingCancellation', () => {
+    let cancellation: TfmDealCancellation = {
+      reason: 'a reason',
+      bankRequestDate: 0,
+      effectiveFrom: 0,
+    };
+
+    const aDealCancellation = (): TfmDealCancellation => ({
+      ...cancellation,
+      bankRequestDate: new Date().valueOf(),
+      effectiveFrom: new Date().valueOf(),
+    });
+
```

Comment:
This is still could be affected by side effects - if one of the functions being tested was written badly & modified `cancellation`, all subsequent tests would be affected by this

Reply:
I don't think we need to change this for a potentially badly written test in the future.

Reply:
Sorry I dont think I was clear, this might not work if `DealCancellationService.processPendingCancellation` is modifying the paramenters, not if future tests are written badly

e.g. if the method  did things like
```
  public static async processPendingCancellation(
    dealId: ObjectId | string,
    cancellation: TfmDealCancellation,
    auditDetails: AuditDetails,
  ): Promise<TfmDealCancellationResponse> {

    // Does some stuff modifying the object
    cancellation.reason = mapReason(cancellation.reason);

    if (!cancellation.auditRecord) {
      cancellation.auditRecord = generateAuditRecordFromAuditDetails(auditDetails)
    }
  }

```

Then we'd be passing in the modified `reason` & `auditRecord` fields in on each subsequent test giving unexpected behaviour.

Reply:
Have updated and simplified the creation of the `cancellation` object